integrated_channels/canvas/client.py

```diff
@@ -51,9 +51,102 @@ def __init__(self, enterprise_configuration):
         self.expires_at = None
 
     def create_course_completion(self, user_id, payload):  # pylint: disable=unused-argument
-        pass
+        learner_data = json.loads(payload)
+        self._create_session()
+
+        # Retrieve the Canvas user ID from the user's edx email (it is assumed that the learner's Edx
+        # and Canvas emails will match).
+        get_user_id_from_email_url = '{url_base}/api/v1/accounts/{account_id}/users?search_term={email_address}'.format(
+            url_base=self.enterprise_configuration.canvas_base_url,
+            account_id=self.enterprise_configuration.canvas_account_id,
+            email_address=user_id
+        )
+        get_users_by_email_response = self.session.get(get_user_id_from_email_url).json()
+        try:
+            canvas_user_id = get_users_by_email_response[0]['id']
+        except (KeyError, IndexError):
+            # Trying to figure out how we should handle errors here- should we catch
+            # them in the in the transmitter? Should we return 400 but not raise?
+            # If we have multiple course completions to post, a raised exception here without
+            # anything else will prevent other transmissions to happen.
+            raise CanvasClientError(
+                "No Canvas user ID found associated with email: {}".format(learner_data['userID'])
+            )
+
+        # With the Canvas user ID, retrieve all courses for the user.
+        get_users_courses_url = '{canvas_base_url}/api/v1/users/{canvas_user_id}/courses'.format(
+            canvas_base_url=self.enterprise_configuration.canvas_base_url,
+            canvas_user_id=canvas_user_id
+        )
+        user_courses_response = self.session.get(get_users_courses_url).json()
```

Comment:
you have loads of these gets not in try/catch blocks, so I'm again curious why you focus on the user id lol

Reply:
I'm a little lost- what's the connection between user ID and catching errors from Canvas? I was thinking I'd add `response.raise_for_status()` after each of the calls.

Reply:
My point was you are making external calls, but asked about using try/catch and raising in one spot and it was just funny/odd to me to have a try/catch only on one external call.

Reply:
yeeeah, there's a point here about consistency. I've done a little clean up to try and make things more consistent.